src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -331,13 +328,77 @@ private void parseJabRefComment(Map<String, String> meta) {
             // custom entry types are always re-written by JabRef and not stored in the file
             dumpTextReadSoFarToString();
         }
+        if (comment.startsWith(MetaData.BIBDESK_FLAG)) {
+            parseBibDeskComment(comment, meta);
+        }
+    }
+
+    /**
+     * Adds BibDesk group entries to the JabRef database
+     * */
+    private void addBibDeskGroupEntriesToJabRefGroups(){
+        for (String groupName: parsedGroups.keySet()) {
+            String[] citationKeys = parsedGroups.get(groupName).split(",");
+            for (String citation : citationKeys){
+                String groupValue = database.getEntryByCitationKey(citation).get().getField(StandardField.GROUPS).orElse(null);
+                if(groupValue == null) { // if the citation does not belong to a group already
+                    database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupName);
+                } else { // if the citation does belong to a group already, we concatinate
+                    groupValue += ","+ groupName;
+                    database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupValue);
+                }
+            }
+        }
+    }
+
+    /**
+     * Parses comment types found in BibDesk, to migrate BibDesk Static Groups to JabRef.
+     * */
+    private void parseBibDeskComment(String comment, Map<String, String> meta) {
+        String xml = comment.substring(MetaData.BIBDESK_FLAG.length() + 1, comment.length() - 1);
+        try {
+            //Build a document to handle the xml tags
+            DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            Document doc = builder.parse(new ByteArrayInputStream(xml.getBytes()));
+            doc.getDocumentElement().normalize();
+
+            NodeList dictList = doc.getElementsByTagName("dict");
+            meta.put("databaseType", "bibtex;");
+            meta.put("grouping", "0 AllEntriesGroup:;");
```

Comment:
Changing this to a putIfAbsent allows for handling multiple group comments in one file